lib/suse/connect/cli.rb

```diff
@@ -60,7 +60,7 @@ def execute! # rubocop:disable MethodLength, CyclomaticComplexity
             " Check #{@options[:url] || 'https://scc.suse.com'} whether your system appears there." \
             ' If it does not, please call SUSEConnect --cleanup and re-register this system.'
           else
-            log.fatal 'Error: Provided registration code is not recognized by registration server.'
+            log.fatal "Error: SCC returned '#{e.message}' (#{e.code})"
```

Comment:
Maybe just a plain `if e.code==401 && System.credentials?` instead of nested `case`?